gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Dataset.java

```diff
@@ -35,89 +30,108 @@
  *
  * <p>Objects of this class are immutable. Operations that modify the dataset like {@link #update}
  * return a new object. To get a {@code Dataset} object with the most recent information use
- * {@link #reload}.
+ * {@link #reload}. {@code Dataset} adds a layer of service-related functionality over
+ * {@link DatasetInfo}.
  * </p>
  */
-public final class Dataset {
+public final class Dataset extends DatasetInfo {
 
-  private final BigQuery bigquery;
-  private final DatasetInfo info;
+  private static final long serialVersionUID = -4272921483363065593L;
 
-  private static class TablePageFetcher implements PageImpl.NextPageFetcher<Table> {
+  private final BigQueryOptions options;
+  private transient BigQuery bigquery;
 
-    private static final long serialVersionUID = 6906197848579250598L;
+  static final class Builder extends DatasetInfo.Builder {
```

Comment:
Are there access issues when making this class package scope but including it in the function signature for public methods like `toBuilder()`?

Even if not, do you think there's a benefit to making the builders for the functional classes public?  As it stands, the javadoc won't have a `Dataset.Builder` page, just a `DatasetInfo.Builder` page.  As a result, the `Dataset.toBuilder()` function links to the `DatasetInfo.Builder`, which has a `build()` method that returns a `DatasetInfo` instead of a `Dataset`.  A user could interpret this to mean that if they call `someDataset.toBuilder().build()`, they will get a `DatasetInfo` and not necessarily a `Dataset`.